File: RB101_Programming_Foundations/small_problems/Easy1/Run2/array_average.rb
# def average(array)
#   array.reduce(&:+) / array.size
# end

def average(array)
  sum = 0
  array.each do |number|
    sum += number
  end
  sum / array.size.to_f
end

puts average([1, 6]) == 3 # integer division: (1 + 6) / 2 -> 3
puts average([1, 5, 87, 45, 8, 8]) == 25
puts average([9, 47, 23, 95, 16, 52]) == 40

File: js210_fundamentals_of_JS_Programming/problem_sets/small_problems/easy4/array_average.js
function average(array) {
  let average = array.reduce((sum, val) => sum + val, 0) / array.length;
  return Math.floor(average);
}
console.log(average([1, 5, 87, 45, 8, 8]));       // 25
console.log(average([9, 47, 23, 95, 16, 52]));    // 40
